### ISSUE TITLE
Gatsby site inconsistently asking for password

### ISSUE THREAD

**Issue by waylonrobert** [robinmoisson/staticrypt#133]
I've got this running on a Gatsby site hosted on Netlify as well as Vercel to rule out anything weird on the Netlify end. The problem I have is that I can inconsistently access protected pages without a password. Sometimes it asks me for a password, other times it does not. A hard refresh almost always triggers the password prompt on the URL. 

- Site: 
  (Netlify): https://631a21da30bc1d007183e00d--transcendent-bonbon-26c472.netlify.app/
  (Vercel): https://my-gatsby-site-eight.vercel.app/

- Protected page:
  (Netlify): https://631a21da30bc1d007183e00d--transcendent-bonbon-26c472.netlify.app/protected/secret
  (Vercel): https://my-gatsby-site-eight.vercel.app/protected/secret

Everything is configured in my `package.json` file located at https://github.com/waylonrobert/my-gatsby-site/blob/main/package.json.

Essentially Netlify and Vercel runs these two commands at deploy, in order:

`npm run build`, which is an alias for the `gatsby build` command that generates the static pages (in this case the secret page located in public/protected). 

`npm run _encrypt`, which runs this command
`"find ./public/protected -type f -name '*.html' -exec staticrypt {} password -o {} \\;"` (and yes I know storing this in a public repo is bad practice, as is using _password_ as the password, but this is just for troubleshooting right now)

Any thoughts on why this is inconsistency is happening? @robinmoisson

**Comment by robinmoisson** [comment]
Hey @waylonrobert thanks for sharing links to your environments. 

I can't reproduce on chrome - everything happens in the browser and in JS so my first thought wouldn't be that it's linked to the hosting provider netlify or vercel. In what browser does this happen to you? If you try on private navigation (which clears cookies, possibly disables extensions) or in a different browser do you have the same thing happening?

Not having to re-enter the password would be what the "Remember-Me" checkbox does - I assume you made sure it wasn't checked? Maybe a browser extension auto-checks it? You can see if the encrypted password is stored in localStorage, which is what the remember-me checkbox does.

**Comment by waylonrobert** [comment]
Thanks @robinmoisson! I've tinkered with this some more and I realized that the problem always occurred whenever I clicked the "Link to secret page" from the main page at https://my-gatsby-site-eight.vercel.app/ or  https://631a21da30bc1d007183e00d--transcendent-bonbon-26c472.netlify.app/. I think it has to do with how Gatsby rehydrates when going to a link using Gatsby's built-in `Link` component (which makes sense because Gatsby has no way of knowing the content is protected). 

So my untested solution would be to use a normal anchor link tag that bypasses Gatsby's link rehydration (though I'd lose some of the benefits, mainly speed) or find some other way of communicating to Gatsby that there is a protected route.
 
Anyway, I don't think this is a problem with staticrypt at all; I'm happy to close this issue?
